src/utils/debounce.js

```diff
@@ -13,13 +14,16 @@
  *  }, ON_WINDOW_RESIZE_DEBOUNCE_DELAY)
  * );
  */
-export const debounce: Function = (origFunc: Function, time: number) => {
+export const debounce: Function = (origFunc: Function, time: number, immediate: boolean = true) => {
   let timeout;
 
   return (...args) => {
+    const context = this;
+    if (immediate && !timeout) origFunc.apply(context, args);
```

Comment:
Do not use inline ifs without curly brackets.
@yairans letw make sure we have lint rule to make sure this is not possible